src/server/middleware/immutable/pushmf.go

```diff
@@ -59,51 +56,21 @@ func handlePush(req *http.Request) error {
 		return nil
 	}
 
-	// match repository ...
-	total, repos, err := repository.Mgr.List(req.Context(), &q.Query{
-		Keywords: map[string]interface{}{
-			"Name": mf.Repository,
-		},
-	})
-	if err != nil {
-		return err
-	}
-	if total == 0 {
-		return nil
-	}
-
 	// match artifacts ...
-	total, afs, err := artifact.Mgr.List(req.Context(), &q.Query{
-		Keywords: map[string]interface{}{
-			"ProjectID":    mf.ProjectID,
-			"RepositoryID": repos[0].RepositoryID,
-		},
+	af, err := artifact.Ctl.GetByReference(req.Context(), mf.Repository, mf.Tag, &artifact.Option{
+		WithTag: true,
 	})
 	if err != nil {
 		return err
 	}
-	if total == 0 {
+	if af == nil {
```

Comment:
`af` will never be `nil` return by `artifact.Ctl.GetByReference` when `err` is not `nil`.

Reply:
ditto